curve25519-dalek/src/scalar.rs

```diff
@@ -779,20 +779,68 @@ impl Scalar {
     /// assert_eq!(scalars[3], Scalar::from(11u64).invert());
     /// # }
     /// ```
+    pub fn batch_invert<const N: usize>(inputs: &mut [Scalar; N]) -> Scalar {
+        let one: UnpackedScalar = Scalar::ONE.unpack().as_montgomery();
+
+        let mut scratch = [one; N];
+
+        Self::batch_invert_internal(inputs, &mut scratch)
+    }
+
+    /// Given a slice of nonzero (possibly secret) `Scalar`s,
+    /// compute their inverses in a batch.
+    ///
+    /// # Return
+    ///
+    /// Each element of `inputs` is replaced by its inverse.
+    ///
+    /// The product of all inverses is returned.
+    ///
+    /// # Warning
+    ///
+    /// All input `Scalars` **MUST** be nonzero.  If you cannot
+    /// *prove* that this is the case, you **SHOULD NOT USE THIS
+    /// FUNCTION**.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use curve25519_dalek::scalar::Scalar;
+    /// # fn main() {
+    /// let mut scalars = [
+    ///     Scalar::from(3u64),
+    ///     Scalar::from(5u64),
+    ///     Scalar::from(7u64),
+    ///     Scalar::from(11u64),
+    /// ];
+    ///
+    /// let allinv = Scalar::batch_alloc_invert(&mut scalars);
+    ///
+    /// assert_eq!(allinv, Scalar::from(3*5*7*11u64).invert());
+    /// assert_eq!(scalars[0], Scalar::from(3u64).invert());
+    /// assert_eq!(scalars[1], Scalar::from(5u64).invert());
+    /// assert_eq!(scalars[2], Scalar::from(7u64).invert());
+    /// assert_eq!(scalars[3], Scalar::from(11u64).invert());
+    /// # }
+    /// ```
```

Comment:
Should I replace this with something like:
```suggestion
    /// This is equivalent to [`batch_invert()`] without memory allocation. See [`batch_invert()`] for more details.
    ///
    /// [`batch_invert()`]: Self::batch_invert
```